dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -151,14 +156,88 @@ public DicomInputStream(InputStream in, String tsuid) throws IOException {
     public DicomInputStream(InputStream in) throws IOException {
         super(in.markSupported() ? in : new BufferedInputStream(in));
         if( in instanceof CloneIt ) originalInput = (CloneIt<InputStream, IOException>) in;
-        this.guessTransferSyntax();
+        this.guessTransferSyntax(DEFAULT_PREAMBLE_LENGTH);
+    }
+
+    public DicomInputStream(InputStream in, int preambleLength) throws IOException {
+        super(ensureMarkSupported(in));
+        guessTransferSyntax(preambleLength);
```

Comment:
This was added as part of the cherry-pick and I'm wondering if it should be re-written to use the constructor above or removed all together. It does not make use of the CloneIt functionality. Do you know what the CloneIt stuff is @mdoris?